extension/user/login.js

```diff
@@ -0,0 +1,53 @@
+const crypto = require('crypto')
+const {EINVALIDCREDENTIALS, ENOTFOUND, customError} = require('../error')
+/**
+ * @typedef {Object} LoginInputArgs
+ * @property {string} strategy
+ * @property {Object} parameters
+ * @property {string} parameters.login
+ * @property {string} parameters.password
+ */
+
+/**
+ * @param {SDKContext} context
+ * @param {LoginInputArgs} input
+ * @param {function} cb
+ */
+module.exports = function (context, input, cb) {
+  // is logged in
+  if (context.meta.userId) {
+    return cb(null, {
+      userId: context.meta.userId
+    })
+  }
+
+  context.storage.extension.get(input.parameters.login, (errUserId, userId) => {
+    if (errUserId) {
+      return cb(errUserId)
+    }
+
+    if (!userId) {
+      return cb(customError(ENOTFOUND, 'User is not found'))
```

Comment:
userId and password should never be separated in a response for security reasons: One would immediately find out, if an email address was registered, or not (even if he doesn't own it).

It should always result in "EINVALIDCREDENTIALS": "The given username and password combination does not exist."